### PR TITLE
integration/crio: create a loop device if the LVM device doesn't exist

### LINKED ISSUE #1575: crio: create a temporal loop device if the lvm device doesn't exist
create a loop device if the default lvm device doesn't exist, that way, testing crio+devicemapper in system without an extra block device will be easier

### PR DESCRIPTION
**Pull request by devimc**
Create a temporal loop device and use as LVM device to test crio+devicemapper,
if the default LVM device doesn't exist. This change is useful
to test crio+devicemapper in systems that don't have an extra block device.

fixes #1575

Signed-off-by: Julio Montes <julio.montes@intel.com>

### CODE REVIEW

File: integration/cri-o/cri-o.sh [hunk 1]
```diff
@@ -62,13 +69,25 @@ IFS=$OLD_IFS
 
 # run CRI-O tests using devicemapper on ubuntu
 if [ "$ID" == "ubuntu" ]; then
+	if [ ! -b "${LVM_DEVICE}" ]; then
+		info "Creating a loop device to use it as LVM device"
+		# create a loop device and use it as lvm device
+		trap cleanup EXIT
+		img_file=devmap.img
+		dd if=/dev/zero of=${img_file} count=10 bs=50M
+		printf "g\nn\n\n\nw\n" | fdisk ${img_file}
```
Comment (grahamwhaley) [inline integration/cri-o/cri-o.sh]: eek - **magic** ;-) 🎩

Comment (devimc) [reply integration/cri-o/cri-o.sh]: heh :smile:

File: integration/cri-o/cri-o.sh [hunk 2]
```diff
@@ -62,13 +69,25 @@ IFS=$OLD_IFS
...
+	fi
+
 	# Block device attached to the VM where we run the CI
 	# If the block device has a partition, cri-o will not be able to use it.
 	export LVM_DEVICE
 	if sudo fdisk -l "$LVM_DEVICE" | grep "${LVM_DEVICE}[1-9]"; then
 		die "detected partitions on block device: ${LVM_DEVICE}. Will not continue"
 	fi
-	export STORAGE_OPTIONS="$DM_STORAGE_OPTIONS"
+	export STORAGE_OPTIONS="$DM_STORAGE_OPTIONS --storage-opt dm.directlvm_device=${LVM_DEVICE}"
```
Comment (jcvenegas) [inline integration/cri-o/cri-o.sh]: Probably not part of the same commit but this should no be part of the `if statement where it checks LVM_DEVICE `

Comment (devimc) [reply integration/cri-o/cri-o.sh]: no, because the `LVM_DEVICE` can be overwritten or it can be the default device (`/dev/vdb`)

**Pull request merged**